GliaWidgets/Sources/Component/OperatorTypingIndicator/OperatorTypingIndicatorView.swift

```diff
@@ -1,73 +1,159 @@
+        color: style.color,
+        duration: animationDuration,
+        beginTime: animationDuration
+    )
+
+    lazy var stack = UIStackView.make(.horizontal, spacing: 3)(
+        dotLeadingView,
+        dotMiddleView,
+        dotTrailingView
+    )
+
     var accessibilityProperties: AccessibilityProperties {
         didSet {
             updateAccessibility()
         }
     }
 
+    private let style: OperatorTypingIndicatorStyle
+
     init(
-        style: OperatorTypingIndicatorStyle,
-        bundleManaging: BundleManaging = .live,
-        accessibilityProperties: AccessibilityProperties
+        accessibilityProperties: AccessibilityProperties,
+        style: OperatorTypingIndicatorStyle
     ) {
-        self.style = style
-        self.bundleManaging = bundleManaging
         self.accessibilityProperties = accessibilityProperties
+        self.style = style
         super.init()
     }
 
-    required init() {
-        fatalError("init() has not been implemented")
-    }
+    required init() { fatalError("init() has not been implemented") }
 
     override func setup() {
         super.setup()
-        let animation = Animation.named(
-            "operator-typing-indicator",
-            bundle: bundleManaging.current()
-        )
-        animationView.animation = animation
-        animationView.backgroundBehavior = .pauseAndRestore
-
-        let keypath = AnimationKeypath(
-            keys: ["**", "Fill 1", "Color"]
-        )
-        let colorValueProvider = ColorValueProvider(style.color.lottieColorValue)
-        animationView.setValueProvider(colorValueProvider, keypath: keypath)
-
-        animationView.contentMode = .scaleAspectFit
-        animationView.loopMode = .loop
-        isAccessibilityElement = true
-        updateAccessibility()
+
+        addSubview(stack)
     }
 
     override func defineLayout() {
         super.defineLayout()
-        addSubview(animationView)
-        animationView.autoSetDimensions(to: kViewSize)
-        animationView.autoPinEdge(toSuperviewEdge: .leading, withInset: kLeftInset)
-        animationView.autoPinEdge(toSuperviewEdge: .top)
-        animationView.autoPinEdge(toSuperviewEdge: .bottom)
+
+        NSLayoutConstraint.activate([
+            stack.leadingAnchor.constraint(equalTo: leadingAnchor),
+            stack.topAnchor.constraint(equalTo: topAnchor),
+            stack.trailingAnchor.constraint(equalTo: trailingAnchor),
+            stack.bottomAnchor.constraint(equalTo: bottomAnchor),
+            dotLeadingView.widthAnchor.constraint(equalTo: dotMiddleView.widthAnchor),
+            dotMiddleView.widthAnchor.constraint(equalTo: dotTrailingView.widthAnchor)
+        ])
     }
 
     private func updateAccessibility() {
         accessibilityLabel = style.accessibility.label.withOperatorName(accessibilityProperties.operatorName)
     }
 }
 
+extension OperatorTypingIndicatorView {
+    final class DotView: BaseView {
+        let color: UIColor
```

Comment:
the same question here

Reply:
fixed - a442c4ff7e7aeb045efb220807aee5e31d8fd6b9